.pre-commit-config.yaml

```
@@ -0,0 +1,130 @@
+    rev: v0.12.4
+    hooks:
+      - id: ruff
+        files: ^surfsense_backend/
+        args: [--fix, --exit-non-zero-on-fix]
+      - id: ruff-format
+        files: ^surfsense_backend/
+
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.17.0
+    hooks:
+    - id: mypy
```

YAMLlint (1.37.1)

[warning] wrong indentation: expected 6 but found 4

(indentation)